src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -405,6 +442,41 @@ CHIP_ERROR SilabsTracer::GetTraceByOperation(TimeTraceOperation aOperation, Muta
     return CHIP_ERROR_NOT_FOUND;
 }
 
+CHIP_ERROR SilabsTracer::GetTraceByOperation(CharSpan & appOperationKey, MutableByteSpan & buffer) const
+{
+    size_t index = 0;
+    ReturnErrorOnFailure(FindAppOperationIndex(appOperationKey, index));
+    return GetTraceByOperation(index + to_underlying(TimeTraceOperation::kNumTraces), buffer);
+}
+
+CHIP_ERROR SilabsTracer::RegisterAppTimeTraceOperation(CharSpan & appOperationKey)
+{
+    VerifyOrReturnError(appOperationKey.size() <= kMaxAppOperationKeyLength, CHIP_ERROR_BUFFER_TOO_SMALL);
+    VerifyOrReturnError(mAppOperationKeyCount < kMaxAppOperationKeys, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(appOperationKey.data() != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    // Verify the key is not already in the array
+    size_t index = 0;
+    VerifyOrReturnError(CHIP_ERROR_NOT_FOUND == FindAppOperationIndex(appOperationKey, index), CHIP_ERROR_INVALID_ARGUMENT);
+
+    memcpy(mAppOperationKeys[mAppOperationKeyCount], appOperationKey.data(), appOperationKey.size());
```

Comment:
Since we are using the CharSpan everywhere, it could be worth while to put `mAppOperationKeys[mAppOperationKeyCount]` in a CharSpan and use the char span copy function that has additionnal size checks.